File: Python/1.py
text = [
  '   + -- + - + -   ',
  '   + --- + - +   ',
  '   + -- + - + -   ',
  '   + - + - + - +   '
];

#ord(): 문자 -> 숫자
#chr(): 숫자 -> 문자

result = [i.strip().replace(' ', '').replace('+', '1').replace('-', '0') for i in text]
''.join(list(map(lambda x : chr(int(x, 2)), result)))

File: JavaScript/1.js
let data = [
  '   + -- + - + -   ',
  '   + --- + - +   ',
  '   + -- + - + -   ',
  '   + - + - + - +   '
];

let result = '';
for (let i of data) {
  result += String.fromCharCode(parseInt(i.replace(/ /g, '').replace(/\+/g, '1').replace(/-/g, '0'), 2)); // 이진수 -> 십진수 -> 문자
}

console.log(result); 